sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -160,13 +160,14 @@ def error(self):
         return self
 
     def __str__(self):
-        return "({}) {}".format(self.code, self.message)
+        return self.message_details()
 
     def message_details(self):
         """Return a detailled string of the error.
         """
         # () -> str
-        error_str = "Code: {}".format(self.code)
+        error_str = "({}) {}".format(self.code, self.message)
+        error_str += "\nCode: {}".format(self.code)
```

Comment:
To make sure the first line includes all info, it seems that 'error_str' is duplicated for 'self.code' and 'self.message'?

Reply:
We duplicate the information because 

error_str = "({}) {}".format(self.code, self.message)

is the old error message and we want to try our best to be backward compatible. (If customer checks the first line of new message, they can get same information as old one).

Reply:
hmm but would this be breaking for anyone who was parsing `error.message_details()`? That was already public right?